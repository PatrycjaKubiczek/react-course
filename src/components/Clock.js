function Clock({ className = "", minutes = 20, seconds = 48 }) {
  if (seconds === 0) {
    seconds = seconds + "0";
  }
  return (
    <h3 className={"Clock " + className} style={{textAlign: "center"}}>
      Pozosta≈Ço {minutes}:{seconds}
    </h3>
  );
}

export default Clock;