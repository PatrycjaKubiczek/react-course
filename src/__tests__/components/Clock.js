import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Clock from "../../components/Clock";

var root = null;
var clockRenderer = null;

describe("<Clock/>", () => {
  describe("when given a minutes and seconds (DOM)", () => {
    beforeEach(() => {
      root = document.createElement("div");
      ReactDOM.render(<Clock minutes={10} seconds={20} />, root);
    });
    it("renders time properly", () => {
      expect(root.childNodes[0].textContent).toMatch(/10:20/);
    });
    it("renders an h3 element", () => {
      expect(root.childNodes[0].nodeName).toEqual("H3");
    });
    it("sets a Clock className an h3 element", () => {
      expect(root.childNodes[0].className).toMatch(/Clock/);
    });
  });

  describe("when given a minutes and seconds (TestRenderer)", () => {
    beforeEach(() => {
      clockRenderer = renderer.create(
        <Clock minutes={10} seconds={20} />
      );
    });
    it("renders properly", () => {
      expect(clockRenderer.toJSON().type).toMatchSnapshot();
    //   expect(clockRenderer.toJSON().type).toEqual("h3");
    //   expect(clockRenderer.toJSON().props).toMatchObject({"className": expect.stringMatching("Clock ")});
    //   expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining([10, 20]));
    });
    xit("renders an h3 element", () => {
      expect(root.childNodes[0].nodeName).toEqual("H3");
    });
    xit("sets a Clock className an h3 element", () => {
      expect(root.childNodes[0].className).toMatch(/Clock/);
    });
  });
  it("sets className to empty string if not given anything else", () => {
    expect(<Clock minutes={10} seconds={20} />).toEqual(
      <Clock className="" minutes={10} seconds={20} />
    );
  });
});


