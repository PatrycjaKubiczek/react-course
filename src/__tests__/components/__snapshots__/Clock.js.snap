// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock/> when given a minutes and seconds (TestRenderer) renders properly 1`] = `"h3"`;
